report/renderer/category-renderer.js

```diff
@@ -484,13 +486,15 @@ export class CategoryRenderer {
    *   ├── …
    *   ⋮
    * @param {LH.ReportResult.Category} category
-   * @param {Object<string, LH.Result.ReportGroup>} [groupDefinitions]
+   * @param {Object<string, LH.Result.ReportGroup>=} groupDefinitions
+   * @param {'PSI'=} environment
+   * @param {{gatherMode: LH.Result.GatherMode}=} options
    * @return {Element}
    */
-  render(category, groupDefinitions = {}) {
+  render(category, groupDefinitions = {}, environment, options) {
```

Comment:
Would be nice to merge `environment` into `options`. Would this be a breaking change?

Reply:
We could do a `environmentOrOptions` and detect with `typeof environmentOrOptions === 'string'`?

Reply:
I think at the moment our contract begins and ends with `ReportGenerator.generateReportHtml`, `renderer.renderReport`, `DOM`, and `ReportUIFeatures`. Even that is being generous, there's an argument that only `lighthouse --output=html` is officially supported.

Changes to CategoryRenderer presently would only break our PSI integration, which can be easily fixed on the next roll to google3 (and tests will prevent a regression).

Reply:
also web.dev, but i don't have any memory of how that works :)